tests/generation/test_utils.py

```diff
@@ -756,38 +777,44 @@ def test_beam_sample_generate_dict_output(self):
                 self.assertIsInstance(output_generate, BeamSampleDecoderOnlyOutput)
 
             self.assertTrue(output_generate.sequences.shape[-1] == max_length)
+            is_vision_model = model_class.__name__ in get_values(MODEL_FOR_VISION_2_SEQ_MAPPING_NAMES)
             self._check_outputs(
-                output_generate, input_ids, model.config, num_return_sequences=beam_kwargs["num_beams"]
+                output_generate,
+                inputs_dict["attention_mask"],
+                model.config,
+                is_vision_model=is_vision_model,
+                num_return_sequences=beam_kwargs["num_beams"],
             )
 
     def test_generate_without_input_ids(self):
-        config, _, _, max_length = self._get_input_ids_and_config()
+        for model_class in self.all_generative_model_classes:
+            config, _, max_length = self._get_input_ids_and_config(model_class)
```

Comment:
added as a loop over all models, so that we can get the `model_class` to infer correct `inputs_dict` format